### PR TITLE
Bug/122 adjust validation for transaction amount

### LINKED ISSUE #122: Adjust validation for transaction amount
It should be allowed to create transaction with amount equal to `0.00`

### PR DESCRIPTION
**Pull request by tomektomek321**
Closes #122 

---

I confirm that (check also those that do not apply to this PR):
- [x] New code is covered with tests and coverage requirements are met
- [x] All tests are passing
- [x] Build is not failing
- [x] Mobile version of new functionality is displayed properly
- [x] All linter issues are resolved
- [x] All conflicts are resolved and branch is rebased on top of the up-to-date parent branch
- [x] All issue requirements are implemented
- [x] At least two reviewers are assigned
- [x] I'm assigned as a assignee
- [x] PR is linked to the related issue

### CODE REVIEW

File: src/app/common/utils/pipes/transaction-amount.pipe.ts [hunk 1]
```diff
@@ -7,9 +7,9 @@ import { WalletTransactionType } from 'src/app/domains/transactions/domains.tran
 export class TransactionAmountPipe implements PipeTransform {
   public transform(value: number, type?: WalletTransactionType): string {
     if(value === 0) {
```
Comment (purbanski-deftcode) [inline src/app/common/utils/pipes/transaction-amount.pipe.ts]:
```suggestion
    if (value === 0) {
      return '0.00';
    }
```

File: src/app/pages/wallets/wallet-details/pages-wallet-details.component.html [hunk 2]
```diff
@@ -51,7 +51,7 @@
       <ng-container matColumnDef="amount">
         <th mat-header-cell *matHeaderCellDef> Amount </th>
         <td mat-cell *matCellDef="let element">
-          <span [ngClass]="element.type === walletTransactionType.Expense ? 'amount--expense' : 'amount--income'">
+          <span [ngClass]="element.amount === 0 ? 'amount--0' : (element.type === walletTransactionType.Expense ? 'amount--expense' : 'amount--income')">
```
Comment (purbanski-deftcode) [inline src/app/pages/wallets/wallet-details/pages-wallet-details.component.html]:
```suggestion
          <span [ngClass]="element.amount === 0 ? 'amount--zero' : (element.type === walletTransactionType.Expense ? 'amount--expense' : 'amount--income')">
```

File: src/app/pages/wallets/wallet-details/pages-wallet-details.component.scss [hunk 3]
```diff
@@ -37,6 +37,7 @@
       .mat-column-amount {
         .amount--income { color: colors.$amount-income; }
         .amount--expense { color: colors.$amount-expense; }
+        .amount--0 { color: colors.$secondary-text; }
```
Comment (purbanski-deftcode) [inline src/app/pages/wallets/wallet-details/pages-wallet-details.component.scss]:
```suggestion
        .amount--zero { color: colors.$secondary-text; }
```

File: src/app/pages/wallets/wallet-details/pages-wallet-details.component.ts
Comment (purbanski-deftcode) [inline src/app/pages/wallets/wallet-details/pages-wallet-details.component.ts:1]: I've also noticed that when adding new transaction it is added to the bottom of the list ignoring the date of the transaction. Maybe you can also fix this as part of this PR instead creating another issue.

File: src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.spec.ts [hunk 4]
```diff
@@ -90,28 +90,42 @@ describe('TransactionEditorComponent', () => {
...
+          });
         });
       });
 
       describe('success', () => {
-        it('should show no validation error if amount has valid format', () => {
-          component.form.get('amount')?.setValue(10.21);
-          component.form.get('amount')?.markAsTouched();
+        describe('should show no validation error if amount has valid format', () => {
+          it('0', () => {
```
Comment (purbanski-deftcode) [inline src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.spec.ts]:
```suggestion
        describe('validation errors', () => {
          it('should not show validation error if amount is equal to 0', () => {
```
Other test descriptions should be changed in similar way. `it` should describe what `should` happen. `describe` is just grouping a bunch of related tests together and label them.

File: src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts [hunk 5]
```diff
@@ -121,7 +121,7 @@ export class PagesWalletTransactionEditorComponent implements OnInit, OnDestroy
     return (control: AbstractControl) => {
       const value: number = parseFloat(control.value as string);
 
-      if(value && value > 0) {
+      if(!isNaN(value) && value >= 0) {
```
Comment (purbanski-deftcode) [inline src/app/pages/wallets/wallet-details/transaction-editor/pages-wallet-transaction-editor.component.ts]:
```suggestion
      if (!isNaN(value) && value >= 0) {
```

**Review by purbanski-deftcode — CHANGES_REQUESTED** [review]

File: src/app/common/utils/sorts/common.util.sort..ts [hunk 6]
```diff
@@ -16,3 +16,24 @@ export function sortAlphabeticallyByProp<T extends object, K extends KeysMatchin
     return propAValue.toLowerCase() < propBValue.toLowerCase() ? -1 : 1;
   });
 }
+
+export function sortByDate<T extends object, K extends KeysMatching<T, Date>>
```
Comment (purbanski-deftcode) [inline src/app/common/utils/sorts/common.util.sort..ts:20]: And where is test for that? :trollface:

File: src/app/common/utils/sorts/common.util.sort..ts [hunk 7]
```diff
@@ -16,3 +16,24 @@ export function sortAlphabeticallyByProp<T extends object, K extends KeysMatchin
...
+(array: T[], prop: K, desc = false): T[] {
```
Comment (purbanski-deftcode) [inline src/app/common/utils/sorts/common.util.sort..ts]:
```suggestion
(array: T[], prop: K, descending = false): T[] {
```

File: src/app/common/utils/sorts/common.util.sort..ts [hunk 8]
```diff
@@ -16,3 +16,24 @@ export function sortAlphabeticallyByProp<T extends object, K extends KeysMatchin
...
+    }
+
+    if (propAValue === propBValue) {
+      return 0;
+    }
+
+    return (desc) ?
+      (propAValue < propBValue ? -1 : 1)
+      :
+      (propAValue < propBValue ? 1 : -1);
```
Comment (purbanski-deftcode) [inline src/app/common/utils/sorts/common.util.sort..ts]: Damn, first time I see such weird ternary, why not just:
```suggestion
    return (desc) ? (propAValue < propBValue ? -1 : 1) : (propAValue < propBValue ? 1 : -1);
```

**Review by purbanski-deftcode — CHANGES_REQUESTED** [review]

**Review by purbanski-deftcode — APPROVED** [review]

**Review by mirekgab — APPROVED** [review]

**Pull request merged**